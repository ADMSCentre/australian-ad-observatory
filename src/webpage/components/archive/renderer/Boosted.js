import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faRocket } from '@fortawesome/free-solid-svg-icons';

const Boosted = ({ boosted }) =>
  boosted ? (
    <div className="btn btn-outline-success">
      <FontAwesomeIcon icon={faRocket} />
      Post was boosted
    </div>
  ) : null;

Boosted.displayName = 'Boosted';

Boosted.propTypes = { boosted: PropTypes.bool };

Boosted.defaultProps = { boosted: false };

export default Boosted;
