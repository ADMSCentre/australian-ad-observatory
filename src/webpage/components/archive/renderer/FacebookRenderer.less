.fb-item {
  padding: 0;
  width: 500px;
  background-color: #fff;
  border-radius: 5px;
  border: 1px solid #ddd;
  margin-top: 25px;
  margin-bottom: 15px;
  margin-right: 15px;

  &:first-child {
    padding: 12px 12px 0 12px;
  }

  .hidden_elem {
    display: none;
  }

  [data-testid="post_message"] {
    padding: 10px 0;
  }

  & > img {
    max-width: 90%;
    border: 0;
    margin: 0;
    padding: 0;
  }

  video {
    display: none;
  }

  div.mtm div div div span div img {
    width: 498px;
    height: auto;
    margin-left: -12px;
    border-top: 1px solid #ccc;
    border-bottom: 1px solid #ccc;
  }

  // Hide divs that have been otherwise rendering as stray periods.
  ._6spk {
    display: none;
  }

  // Don't allow user to click buttons or links.
  // a,
  // button {
  //   // pointer-events: none;
  // }

  h5 {
    font-size: 14px; // Match size of source bar w/ Facebook's size
    margin-bottom: 0; // Bring timestamp closer to the source bar title
  }

  // Add "arrow" in place of "to" in source bar.
  ._gb8.img::after {
    content: ">";
    font-style: normal;
  }

  // Remove word "to" in source bar.
  ._gb8 u {
    display: none;
  }

  ._6m6 {
    padding-top: 2px;
    margin-top: 3px;
    display: block;
    float: left;
    padding-bottom: 10px;

    a {
      color: #1d2129;
      font-size: 16px;
      font-weight: 600;

      &:hover {
        text-decoration: none;
        color: #1d2129;
      }
    }
  }

  // Bring author names inline with profile pics.
  ._6a._6b {
    display: inline-block;
    height: 20px !important;

    a {
      color: #385898;
      font-size: 14px;
      font-weight: 600;
      font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
      -webkit-font-smoothing: subpixel-antialiased;
    }
  }

  ._ohe ._6a {
    display: block;
    float: right;
    padding-left: 18px;
    height: 20px;
    vertical-align: middle;

    a {
      border: 1px solid rgb(190, 195, 201);
      background-color: rgb(238, 240, 242);
      color: #4b4f56;
      padding: 5px 8px;
      border-radius: 2px;

      &:hover {
        text-decoration: none;
      }
    }
  }

  ._6m7 {
    margin-top: 3px;
    color: #606770;
    font-size: 14px;
    line-height: 20px;
    word-break: break-word;
    display: none;
    overflow-x: hidden;
  }

  div[data-testid="post_message"] p {
    color: #1d2129;
    font-size: 14px;
    font-weight: 400;
    margin-bottom: 5px;
  }

  .accessible_elem {
    display: none;
  }

  ._4r_y {
    height: 2px;
  }

  // Hide inconsequential artifacts.
  ._52c6 {
    display: none;
  }

  div ._3x-2 {
    margin: 0;
    padding: 0;
  }

  // Hide buttons.
  button {
    display: none;
  }

  // Profile pic wrapper.
  ._5pb8._1yz2._8o._8s.lfloat._ohe,
  ._38vo {
    display: block;
    float: left;
    width: 32px;
    margin-right: 8px;
    border-radius: 16px;
    overflow: hidden;
  }

  ._38vo {
    max-height: 32px;
  }

  ._38vo img {
    max-height: 32px;
  }

  ._5x46,
  ._3dp ._29k {
    /* Profile name wrapper */
    margin-bottom: 6px;
  }

  ._3ekx {
    display: block;
  }

  ._6lz {
    padding-top: 10px;
    color: #606770;
    font-size: 12px;
  }

  .uiScaledImageContainer {
    max-width: 100%;
    width: auto;
    height: auto;
  }

  .scaledImageFitWidth {
    // max-width: 100%;
    // width: auto;
    // height: auto;
  }

  .ellipsis {
    line-height: 14px;
    text-transform: uppercase;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  .card-title {
    display: none;
  }

  .card-subtitle {
    display: none;
  }

  &.fb-publicUserPost {
    // background-color: #e1edec;
    div.mtm div div div span div div {
      display: none;
    }
  }

  &.fb-publicPagePost {
    // background-color: #e1edec;
    div.mtm div div div span div div {
      display: none;
    }
  }

  &.fb-sponsoredPost {
    // background-color: #f1edec;

    a[role="button"] {
      //border: 1px solid;
      border-radius: 2px;
      box-sizing: content-box;
      font-size: 12px;
      -webkit-font-smoothing: antialiased;
      font-weight: bold;
      justify-content: center;
      padding: 0 8px;
      position: relative;
      text-align: center;
      text-shadow: none;
      vertical-align: middle;
    }
  }

  ul.uiList {
    margin-left: -52px;
  }

  ul.uiList li {
    display: none;
  }

  ul.uiList li:first-child {
    display: inline-block;
    border-top: 1px solid #ccc;
    width: 100%;
  }

  ul.uiList li img {
    width: 498px;
    height: auto;
  }

  ._5eut,
  ._42gt {
    display: none;
  }

  ._4lpe,
  ._3htz,
  ._5mly,
  ._40ja,
  ._45oh {
    display: none;
  }
}

.targeting {
  margin-left: 5px;
  width: 500px;

  h4 {
    font-size: 12px;
    font-weight: 600;
    margin: 0;
    padding: 0;
  }

  .reason {
    margin-right: 10px;

    .reason-name {
      padding-left: 5px;
    }
  }
}
