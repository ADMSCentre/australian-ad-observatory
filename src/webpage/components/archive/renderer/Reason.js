import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faUsers,
  faGlobe,
  faTransgenderAlt,
  faMapMarker,
  faDotCircle
} from '@fortawesome/free-solid-svg-icons';

const REASONS = {
  WAISTUICustomAudienceType: {
    name: 'Custom Audience',
    icon: faUsers
  },
  WAISTUILocaleType: {
    name: 'Locale',
    icon: faGlobe
  },
  WAISTUIAgeGenderType: {
    name: 'Age and Gender',
    icon: faTransgenderAlt
  },
  WAISTUILocationType: {
    name: 'Location',
    icon: faMapMarker
  }
};

const Reason = ({ reason }) =>
  REASONS[reason] ? (
    <div className="reason badge badge-primary">
      <FontAwesomeIcon icon={REASONS[reason].icon} />
      <span className="reason-name">{REASONS[reason].name}</span>
    </div>
  ) : (
    <div className="reason badge badge-primary">
      <FontAwesomeIcon icon={faDotCircle} />
      <span className="reason-name">{reason}</span>
    </div>
  );

Reason.displayName = 'Reason';

Reason.propTypes = {
  reason: PropTypes.string.isRequired
};

export default Reason;
