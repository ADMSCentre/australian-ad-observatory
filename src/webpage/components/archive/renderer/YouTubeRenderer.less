.yt-item {
  &.yt-instreamVideoAd {
    background-color: #e1edec;
  }

  &.yt-videowallIframeCompanionAd {
    background-color: #f1edec;
  }

  &.yt-actionCompanionAd {
    background-color: #e2edec;
  }

  &.yt-imageCompanionAd {
    background-color: #f2edec;
  }

  &.yt-adActionInterstitial {
    background-color: #e3edec;
  }

  &.yt-invideoOverlayAd {
    background-color: #f3edec;
  }
}
