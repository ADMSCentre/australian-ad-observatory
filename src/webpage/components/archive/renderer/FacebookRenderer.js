import React from 'react';
import PropTypes from 'prop-types';

import Reason from './Reason';
import Boosted from './Boosted';

import './FacebookRenderer.less';

const itemShape = {
  id: PropTypes.string.isRequired,
  itemId: PropTypes.string.isRequired,
  itemType: PropTypes.string.isRequired,
  payload: PropTypes.shape({
    contentHtml: PropTypes.string.isRequired
  }).isRequired
};

const AdTargeting = ({ item }) => {
  if (!item.payload.adTargetingData) return null;

  const {
    data: { waist_targeting_data: targetingData, waist_is_marketplace_boosted_listing: boosted }
  } = item.payload.adTargetingData;
  const targetingReasons = targetingData.map(({ __typename }) => __typename);

  return (
    <div className="targeting">
      <h4>Ad Targeting</h4>
      <Boosted boosted={boosted} />
      {targetingReasons.map(reason => (
        <Reason key={reason} reason={reason} />
      ))}
    </div>
  );
};
AdTargeting.displayName = 'Ad targeting';

AdTargeting.propTypes = {
  item: PropTypes.shape(itemShape).isRequired
};

const FacebookRenderer = ({ item }) => {
  const {
    id,
    itemId: _itemId,
    itemType,
    payload: { contentHtml: html }
  } = item;

  const classes = `ati-item fb-item fb-${itemType}`;

  return (
    <div className="archive-item">
      <div id={id} className={classes} dangerouslySetInnerHTML={{ __html: html }} />
      <AdTargeting item={item} />
    </div>
  );
};

FacebookRenderer.propTypes = {
  item: PropTypes.shape(itemShape).isRequired
};

export default FacebookRenderer;
