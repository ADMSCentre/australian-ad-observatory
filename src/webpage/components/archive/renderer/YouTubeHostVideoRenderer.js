import React from 'react';
import PropTypes from 'prop-types';
import Player from 'youtube-embed-video';

import Card from '../../Card';

import './YouTubeRenderer.less';

const YouTubeHostVideoRenderer = ({ item }) => {
  const classes = 'ati-item yt-item';

  return (
    <Card title={item.title} subtitle={item.author}>
      <div className={classes}>
        <Player videoId={item.id} suggestions={false} enhancedPrivacy={true} size="small" />
      </div>
    </Card>
  );
};

YouTubeHostVideoRenderer.displayName = 'YouTubeHostVideoRenderer';

YouTubeHostVideoRenderer.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    channelId: PropTypes.string,
    url: PropTypes.string,
    author: PropTypes.string,
    title: PropTypes.string
  }).isRequired // determine a precise shape for this.
};

export default YouTubeHostVideoRenderer;
