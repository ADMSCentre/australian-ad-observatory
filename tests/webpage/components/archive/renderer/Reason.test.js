import React from 'react';
import renderer from 'react-test-renderer';
import Reason from 'webpage/components/archive/renderer/Reason';

let tree;

[
  'WAISTUICustomAudienceType',
  'WAISTUILocaleType',
  'WAISTUIAgeGenderType',
  'WAISTUILocationType',
  'other'
].forEach(reason => {
  describe(`can render ${reason}`, () => {
    beforeAll(() => {
      tree = renderer.create(<Reason reason={reason} />);
    });

    it('renders correctly', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
