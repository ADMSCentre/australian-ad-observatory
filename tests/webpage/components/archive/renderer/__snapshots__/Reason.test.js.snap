// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render WAISTUIAgeGenderType renders correctly 1`] = `
<div
  className="reason badge badge-primary"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-transgender-alt fa-w-15 "
    data-icon="transgender-alt"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 480 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M468 0h-79c-10.7 0-16 12.9-8.5 20.5l16.9 16.9-80.7 80.7C294.5 104.1 268.2 96 240 96c-28.2 0-54.5 8.1-76.7 22.1l-16.5-16.5 19.8-19.8c4.7-4.7 4.7-12.3 0-17l-28.3-28.3c-4.7-4.7-12.3-4.7-17 0l-19.8 19.8-19-19 16.9-16.9C107.1 12.9 101.7 0 91 0H12C5.4 0 0 5.4 0 12v79c0 10.7 12.9 16 20.5 8.5l16.9-16.9 19 19-19.8 19.8c-4.7 4.7-4.7 12.3 0 17l28.3 28.3c4.7 4.7 12.3 4.7 17 0l19.8-19.8 16.5 16.5C104.1 185.5 96 211.8 96 240c0 68.5 47.9 125.9 112 140.4V408h-36c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h36v28c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-28h36c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-36v-27.6c64.1-14.6 112-71.9 112-140.4 0-28.2-8.1-54.5-22.1-76.7l80.7-80.7 16.9 16.9c7.6 7.6 20.5 2.2 20.5-8.5V12c0-6.6-5.4-12-12-12zM240 320c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="reason-name"
  >
    Age and Gender
  </span>
</div>
`;

exports[`can render WAISTUICustomAudienceType renders correctly 1`] = `
<div
  className="reason badge badge-primary"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-users fa-w-20 "
    data-icon="users"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="reason-name"
  >
    Custom Audience
  </span>
</div>
`;

exports[`can render WAISTUILocaleType renders correctly 1`] = `
<div
  className="reason badge badge-primary"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-globe fa-w-16 "
    data-icon="globe"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 496 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="reason-name"
  >
    Locale
  </span>
</div>
`;

exports[`can render WAISTUILocationType renders correctly 1`] = `
<div
  className="reason badge badge-primary"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-map-marker fa-w-12 "
    data-icon="map-marker"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="reason-name"
  >
    Location
  </span>
</div>
`;

exports[`can render other renders correctly 1`] = `
<div
  className="reason badge badge-primary"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-dot-circle fa-w-16 "
    data-icon="dot-circle"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="reason-name"
  >
    other
  </span>
</div>
`;
